### DISCUSSION TITLE
If .ssh/config present backups don't start

### DISCUSSION THREAD

**Discussion by cerebrux** [borgbase/vorta#842] [General]
**Describe the bug**
The Vorta Backup does not work if there is a `config` file with settings in `.ssh/`

**To Reproduce**
1. Go to Vorta 
2. Click on Backup now
3. See error : ERROR: Connection closed by remote host. Is borg working on server ?

Then to workaround it do:

1. Move or remove your config eg. `mv .ssh/config .ssh/myconfig-backup`
2. Go to Vorta
3. Click on Backup now
4. Backups now working

**Desktop (please complete the following information):**
- OS: Ubuntu 20.10
- Vorta version: v0.7.1
- Borg version: 1.1.13
- Installed from: Ubuntu Software (flatpak-system)

**Additional context**
When the error occurs, logs are displaying : 
```
vorta.borg.borg_thread - INFO - Running command /app/bin/borg create --list --progress --info --log-json --json --filter=AM -C lz4  /home/user/yolo
vorta.borg.borg_thread - WARNING - Remote: Bad owner or permissions on /home/user/.ssh/config
vorta.borg.borg_thread - ERROR - Connection closed by remote host. Is borg working on the server?
```
When I apply the workaround, logs are displaying:
```
vorta.borg.borg_thread - DEBUG - Using VortaSecretStorageKeyring keyring to store passwords.
asyncio - DEBUG - Using selector: EpollSelector
root - DEBUG - Found 1 passwords matching repo URL.
vorta.borg.borg_thread - INFO - Running command /app/bin/borg create --list --progress --info --log-json --json --filter=AM -C lz4 [....] /home/user/yolo
vorta.borg.borg_thread - INFO - Creating archive at "[....]"
vorta.borg.borg_thread - INFO - Remote: Storage quota [....]
```
I'm not sure, but it seems related to https://github.com/borgbase/vorta/issues/526

**Comment by samuel-w** [comment]
I have an ssh config and it works fine..

**Comment by samuel-w** [comment]
The error message seems to be permission related. Check your ssh config permissions.

**Comment by m3nu** [comment]
Same here. Had my `~/.ssh/config` forever and never an issue with Borg. Maybe the issue is due to a specific setting **in** your config?

**Comment by cerebrux** [comment]
This is for all users out there that may or may not encounter the same problem.

What I did to test and finally fix my problem :

- Remove (or rename ) any file named `config` from `.ssh`
- Launch Vorta, click "Backup Now" and backups work as expected.

Now let's "brake" it: 

- Create an empty `config` file inside .ssh/ with your regular user eg. `touch .ssh/config`
- Launch Vorta, click "Backup Now" and you get the message "ERROR: Connection closed by remote host. Is borg working on server?".
- Check log files where you get : `WARNING - Remote: Bad owner or permissions on /home/user/.ssh/config`
- Check permissions of the empty `config` file : `-rw-rw-r-- 1 user user  config`

Now let's "fix" it:

- Even though the user had the permission to read and edit the file, this is not sufficient. I had to remove the permissions from others. In other words, I needed to have `rw` for user only permissions on config. 
- To fix permissions run `chmod 600 .ssh/config`
- Launch Vorta, click "Backup Now" and backups work as expected.

Should the message ""ERROR: Connection closed by remote host. Is borg working on server" also include the warning `WARNING - Remote: Bad owner or permissions on /home/user/.ssh/config` ?

**Comment by Hofer-Julian** [comment]
> Should the message ""ERROR: Connection closed by remote host. Is borg working on server" also include the warning WARNING - Remote: Bad owner or permissions on /home/user/.ssh/config 

Yes, I think that would be for the best

**Comment by samuel-w** [comment]
So the issue to solve here is displaying the ssh error messages, instead of overwriting them.
Example json output:
```
{"type": "log_message", "time": 1607677712.879403, "message": "Remote: Warning: Identity file /home/user/.ssh/keyname not accessible: No such file or directory.", "levelname": "WARNING", "name": "root"}
{"type": "log_message", "time": 1607677712.8797798, "message": "Remote: Bad owner or permissions on /home/user/.ssh/config", "levelname": "WARNING", "name": "root"}
{"type": "log_message", "time": 1607677712.883788, "message": "Connection closed by remote host. Is borg working on the server?", "levelname": "ERROR", "name": "borg.archiver", "msgid": "ConnectionClosedWithHint"}
```

**Comment by mwmcode** [comment]
Thanks @cerebrux the workaround _eventually_ worked. 


I think this should have a higher priority especially since the error message is not clear.
I do use the ssh config file:
```sh
Host ssh.dev.azure.com
  User git
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/work_azure_rsa
  IdentitiesOnly yes

Host *
  User git
  IdentityFile ~/.ssh/id_rsa
  IdentitiesOnly yes
  AddKeysToAgent yes
  UseKeychain yes

```

**Reply by mwmcode** [reply]
It works when I add an entry for borgbase
```shell
Host MY.repo.borgbase.com
  AddKeysToAgent no
  UseKeychain no
  IdentityFile ~/.ssh/borgbase
  IdentitiesOnly yes
```

> Need to remember to update/add the entry when connecting to a new borgbase repo

**Comment by JocelynDelalande** [comment]
What is weird is that with same ~/.ssh/config (world readable)

- SSH works fine
-  borg in CLI works fine
- vorta fails (with the wrong message in UI, as already said)

So appart the message being unclear, maybe vorta is trying to oversmart something ?

**Reply by m3nu** [reply]
Please provide the log.

Does the issue happen in the current `master` branch on Github? Your error is probably already fixed via #789

**Reply by JocelynDelalande** [reply]
Log was the same as @cerebrux posted :

```
vorta.borg.borg_thread - WARNING - Remote: Bad owner or permissions on /home/user/.ssh/config
vorta.borg.borg_thread - ERROR - Connection closed by remote host. Is borg working on the server?
```

version : v0.7.2 with flatpak

**Reply by m3nu** [reply]
This error is unrelated to Vorta. You can try the solution described here: https://serverfault.com/questions/253313/ssh-returns-bad-owner-or-permissions-on-ssh-config

**Reply by SerhiyMakarenko** [reply]
I can confirm the same behavior described by @JocelynDelalande. I have .ssh/config managed by Nix package manager. It's just symlink to the generated file stored under the hood of the Nix package manager:

```
[15:47] serhiy_makarenko [/home/serhiy_makarenko/.ssh] > ls -lah
total 36K
drwx------  4 serhiy_makarenko serhiy_makarenko 4,0K мар 14 16:00 ./
drwxr-xr-x 27 serhiy_makarenko serhiy_makarenko 4,0K мар 14 11:42 ../
lrwxrwxrwx  1 serhiy_makarenko serhiy_makarenko   74 мар 14 15:47 config -> /nix/store/8g3g9g0zmydz1mrg2dcrxmcpl49nqh2i-home-manager-files/.ssh/config
-rw-------  1 serhiy_makarenko serhiy_makarenko  452 мар 14 10:15 id_ed25519
-rw-------  1 serhiy_makarenko serhiy_makarenko  132 мар 14 10:15 id_ed25519.pub
-rw-------  1 serhiy_makarenko serhiy_makarenko 1,9K мар 14 10:18 known_hosts
drwxr-xr-x  2 serhiy_makarenko serhiy_makarenko 4,0K мар 14 14:45 rsa/
[16:02] serhiy_makarenko [/home/serhiy_makarenko/.ssh] >
```
I'm not able to change permissions on this file. And the result is weird:

- SSH works fine
- borg in CLI works fine
- Vorta fails (with the wrong message in UI about borg not running on the host)

The Vorta log output is the same:
```
vorta.borg.borg_thread - WARNING - Remote: Bad owner or permissions on /home/user/.ssh/config
vorta.borg.borg_thread - ERROR - Connection closed by remote host. Is borg working on the server?
```
I was successful to make a workaround -- add shell commands before and after backup:
```
mv .ssh/config .ssh/config-backup
mv .ssh/config-backup .ssh/config
```
But this is not a very good solution since during the long backup run I will be not able to use my ssh configuration.

I will be very appreciative if we can find a solution to this issue.

Thank you.

**Reply by m3nu** [reply]
The error you are seeing is from SSH, not Vorta. Your workaround may mask some other permission issue. Maybe Vorta runs as different user? Asking because the users/home folders in your logs are all different. (`/home/user/.ssh/config`, `/home/serhiy_makarenko/.ssh`)

**Reply by SerhiyMakarenko** [reply]
Thank you for the prompt replay, @m3nu.

I have checked the logs, the output the following:
```
2021-03-14 10:52:51,809 - vorta.borg.borg_thread - WARNING - Remote: Bad owner or permissions on /home/serhiy_makarenko/.ssh/config
2021-03-14 10:52:51,810 - vorta.borg.borg_thread - ERROR - Connection closed by remote host. Is borg working on the server?
```
Probably, when I wrote my comment I just copy a reply from @JocelynDelalande because it looks similar to records in my logs.
According to the htop Volta running from my user (I have installed it from flatpak repo):
![htop -home-serhiy_makarenko_001](https://user-images.githubusercontent.com/9378038/111082562-9a06b080-8511-11eb-8f6c-7c7ba7bc3a8e.png)
Vorta use some python libs to interact ssh?

Thanks, Serhiy

**Comment by myfirstnameispaul** [comment]
I just experienced the same problem with flatpak install. I uninstalled, then installed with `pip` and no problems connecting. Finally found this discussion after some searching, set `600` on `config` and flatpak version worked immediately.

This is in Linux Mint 20 with 0.8.6.
